eventgrid/data-plane/src/main/java/com/microsoft/azure/eventgrid/customization/EventGridSubscriber.java

```diff
@@ -0,0 +1,163 @@
+        }
+        if (!this.eventTypeToEventDataMapping.containsKey(eventType.toLowerCase())) {
+            return null;
+        }
+        return this.eventTypeToEventDataMapping.get(eventType.toLowerCase());
+    }
+
+    /**
+     * Removes the mapping with the given eventType.
+     *
+     * @param eventType the event type name
+     * @return true if the mapping exists and removed, false if mapping does not exists
+     */
+    @Beta
+    public boolean removeCustomEventMapping(String eventType) {
+        if (eventType == null || eventType.isEmpty()) {
+            throw new IllegalArgumentException("eventType parameter is required and cannot be null or empty");
+        }
+        if (!this.eventTypeToEventDataMapping.containsKey(eventType.toLowerCase())) {
+            return false;
+        }
+        this.eventTypeToEventDataMapping.remove(eventType.toLowerCase());
+        return true;
+    }
+
+    /**
+     * Checks an event mapping with the given eventType exists.
+     *
+     * @param eventType the event type name
+     * @return true if the mapping exists, false otherwise
+     */
+    @Beta
+    public boolean containsEventMappingFor(String eventType) {
+        if (eventType == null || eventType.isEmpty()) {
+            throw new IllegalArgumentException("eventType parameter is required and cannot be null or empty");
+        }
+        return this.eventTypeToEventDataMapping.containsKey(eventType.toLowerCase());
+    }
+
+    /**
+     * De-serialize the events in the given requested content using default de-serializer.
+     *
+     * @param requestContent the request content in string format
+     * @return De-serialized events.
+     *
+     * @throws IOException
+     */
+    @Beta
+    public EventGridEvent[] DeserializeEventGridEvents(final String requestContent) throws IOException {
+        return this.DeserializeEventGridEvents(requestContent, this.defaultSerializerAdapter);
+    }
+
+    /**
+     * De-serialize the events in the given requested content using the provided de-serializer.
+     *
+     * @param requestContent the request content in string format
+     * @param serializerAdapter the de-serializer
+     * @return e-serialized events.
+     * @throws IOException
+     */
+    @Beta
+    public EventGridEvent[] DeserializeEventGridEvents(final String requestContent, final SerializerAdapter<ObjectMapper> serializerAdapter) throws IOException {
+        EventGridEvent[] eventGridEvents = serializerAdapter.<EventGridEvent[]>deserialize(requestContent, EventGridEvent[].class);
+        for (EventGridEvent receivedEvent : eventGridEvents) {
+            if (receivedEvent.data() == null) {
+                continue;
+            } else {
+                final String dataStr = serializerAdapter.serializeRaw(receivedEvent.data());
+                final String eventType = receivedEvent.eventType();
+                if (SystemEventTypeMappings.containsMappingFor(eventType)) {
```

Comment:
Nit: can we do something like if (SystemEventTypeMappings.containsMappingFor(eventType) { eventDataType = SystemEventTypeMappings.getMapping(EventType) } else { eventDataType = getCustomEventMapping(eventType);} and then actually call deserialize & setEventData just once outside the if/else block.

Reply:
1. sounds good, will update the code as suggested. 
2. yes, so special handing required for primitives and arrays. we do have tests covering these scenarios.

Reply:
The primitive and array tests are:

https://github.com/Azure/azure-sdk-for-java/blob/42e2c97bb107e83afbe2b5f24003725ab72e3772/eventgrid/data-plane/src/test/java/com/microsoft/azure/eventgrid/customization/CustomizationTests.java#L77

https://github.com/Azure/azure-sdk-for-java/blob/42e2c97bb107e83afbe2b5f24003725ab72e3772/eventgrid/data-plane/src/test/java/com/microsoft/azure/eventgrid/customization/CustomizationTests.java#L95